File: csharp/Gherkin/Ast/ScenarioOutline.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace Gherkin.Ast
{
    public class ScenarioOutline : ScenarioDefinition
    {
        public IEnumerable<Examples> Examples { get; private set; }

        public ScenarioOutline(Tag[] tags, Location location, string keyword, string title, string description, Step[] steps, Examples[] examples) 
            : base(tags, location, keyword, title, description, steps)
        {
            Examples = examples;
        }
    }
}

File: java/src/main/java/gherkin/ast/ScenarioOutline.java
package gherkin.ast;

import java.util.Collections;
import java.util.List;

public class ScenarioOutline extends ScenarioDefinition {
    private final List<Examples> examples;

    public ScenarioOutline(List<Tag> tags, Location location, String keyword, String name, String description, List<Step> steps, List<Examples> examples) {
        super(tags, location, keyword, name, description, steps);
        this.examples = Collections.unmodifiableList(examples);
    }

    public List<Examples> getExamples() {
        return examples;
    }

    @Override
    public void accept(Visitor visitor) {
        for (Step step : getSteps()) {
           step.accept(visitor);
        }
        for (Examples examples : this.examples) {
            examples.accept(visitor);
        }
        visitor.visitScenarioOutline(this);
    }
}
